web/channels/game_channel.ex

```diff
@@ -1,12 +1,15 @@
 defmodule Pirates.GameChannel do
   use Phoenix.Channel
-  alias Pirates.GameServer.State
+  alias Pirates.GameServer.Instance.State
+  require Logger
 
   def join("game:lobby", _message, socket) do
-    {:ok, table} = Pirates.GameServer.register()
-    socket = assign(socket, :table, table)
-    socket = assign(socket, :id, System.unique_integer([:positive]))
-    {:ok, socket}
+    {:ok, server} = Pirates.GameServer.Manager.get_available_server()
+    {:ok, table} = Pirates.GameServer.Instance.register(server)
+    
+    {:ok, socket |>
+      assign(:table, table) |> 
+      assign(:id, System.unique_integer([:positive]))}
```

Comment:
Your complaint isn't in line with several other bits of code you've already written, plus it's easier to read with the temp variable.